exporters/datadog/lib/opentelemetry/exporters/datadog/exporter/span_encoder.rb

```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'ddtrace/span'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/app_types'
+require 'ddtrace/contrib/redis/ext'
+require 'opentelemetry/trace/status'
+require 'ddtrace/distributed_tracing/headers/headers'
+
+module OpenTelemetry
+  module Exporters
+    module Datadog
+      class Exporter
+        # @api private
+        class SpanEncoder
+          ENV_KEY = 'env'
+          VERSION_KEY = 'version'
+          DD_ORIGIN = '_dd_origin'
+          AUTO_REJECT = 0
+          AUTO_KEEP = 1
+          USER_KEEP = 2
+          SAMPLE_RATE_METRIC_KEY = '_sample_rate'
+          SAMPLING_PRIORITY_KEY = '_sampling_priority_v1'
+          TRUNCATION_HELPER = ::Datadog::DistributedTracing::Headers::Headers.new({})
+
+          INSTRUMENTATION_SPAN_TYPES = {
+            'OpenTelemetry::Adapters::Ethon' => ::Datadog::Ext::HTTP::TYPE_OUTBOUND,
+            'OpenTelemetry::Adapters::Excon' => ::Datadog::Ext::HTTP::TYPE_OUTBOUND,
+            'OpenTelemetry::Adapters::Faraday' => ::Datadog::Ext::HTTP::TYPE_OUTBOUND,
+            'OpenTelemetry::Adapters::Net::HTTP' => ::Datadog::Ext::HTTP::TYPE_OUTBOUND,
+            'OpenTelemetry::Adapters::Rack' => ::Datadog::Ext::HTTP::TYPE_INBOUND,
+            'OpenTelemetry::Adapters::Redis' => ::Datadog::Contrib::Redis::Ext::TYPE,
+            'OpenTelemetry::Adapters::RestClient' => ::Datadog::Ext::HTTP::TYPE_OUTBOUND,
+            'OpenTelemetry::Adapters::Sidekiq' => ::Datadog::Ext::AppTypes::WORKER,
+            'OpenTelemetry::Adapters::Sinatra' => ::Datadog::Ext::HTTP::TYPE_INBOUND
+          }.freeze
+
+          def translate_to_datadog(otel_spans, service, env = nil, version = nil, tags = nil) # rubocop:disable Metrics/AbcSize
+            datadog_spans = []
+
+            default_tags = get_default_tags(tags) || {}
+
+            otel_spans.each do |span|
+              trace_id, span_id, parent_id = get_trace_ids(span)
+              span_type = get_span_type(span)
+
+              datadog_span = ::Datadog::Span.new(nil, span.name,
+                                                 service: service,
```

Comment:
It looks like it will only be possible to have one `service` name for the whole application being instrumented. This might limit the usage, given Datadog's backend can separate different services into distinct buckets for aggregation.
Is this limitation acceptable here?

Reply:
I think this is by design, or at least, this is how python's exporter approaches service naming. i am going to check to see whether this was an oversight or not, as I suppose you could technically leverage the `span_type` field to format the different instrumented library service names as`<service_name>-<span_type_shortname>` so it would look like `my_service-redis` `my_service-net` etc etc